ping/ping.go

```diff
@@ -0,0 +1,212 @@
+package ping
+
+import (
+	"fmt"
+	"math/rand"
+	"sort"
+	"time"
+
+	"github.com/go-ping/ping"
+
+	"github.com/vertcoin-project/one-click-miner-vnext/logging"
+	"github.com/vertcoin-project/one-click-miner-vnext/networks"
+	"github.com/vertcoin-project/one-click-miner-vnext/util"
+)
+
+//To add your node to the list simply add it to the nodes variable with the correct syntax and change the index number to relect the current number of nodes
+//Your node can have a combined max fee of currently 2%, it will not be selected if it doesn't meet that requirement
+//Your node needs to be pingable (Allow ICMP pings)
+//Your node should be live 24/7 to avoid disrupting miners.
+
+var nodes = []string{
+	"fr1.vtconline.org",
+	"p2proxy.vertcoin.org",
+	"mindcraftblocks.com",
+	"vtc-fl.javerity.com",
+	"vtc-ca.javerity.com",
+	"vtc.p2pminers.nl",
+	"p2p-usa.xyz",
+	"p2p-ekb.xyz",
+	"asia.p2p-spb.xyz",
+	"p2p-spb.xyz",
+	"p2p-south.xyz",
+	"siberia.mine.nu",
+}
+var NodeResults = make(map[time.Duration]string)
+
+type Conditions struct {
+	MaxFee      float64
+	MaxMiners   int
+	MaxNextPing time.Duration
+	PingPackets int
+	PingTimeout time.Duration
+}
+
+var Set = Conditions{
+	MaxFee:      2,
+	MaxMiners:   35,
+	MaxNextPing: 30 * time.Millisecond,
+	PingPackets: 2,
+	PingTimeout: 2000000000,
+}
+
+type SelectedNode struct {
+	P2PoolStratum string
+	P2PoolURL     string
+}
+
+var Node SelectedNode
+
+func GetNode(testnet bool) {
+	if testnet {
+		Node = SelectedNode{
+			P2PoolStratum: networks.Active.P2ProxyStratum,
+			P2PoolURL:     networks.Active.P2ProxyURL,
+		}
+	} else {
+		Node = SelectedNode{
+			P2PoolStratum: getClosestNodeStratum(),
+			P2PoolURL:     getClosestNodeURL(),
+		}
+	}
+}
+
+var selectedNode string = selector()
```

Comment:
Why are you doing this on init, while still having a function `GetNode(testnet bool)`? I would suggest merging these two functions into one and just do what `selector()` does before setting the Node.

Also, `GetNode()` might not be the best name for this  method, perhaps `DiscoverBestNode` ?

Reply:
Artifact of the previous version. Will merge these. I believe this will also solve the limited logging that occurs when pinging/selecting nodes.

Name noted.

Reply:
As I suspected this also fixed the limited logging.